src/weights.jl

```diff
@@ -53,9 +57,34 @@ Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values),
     isfinite(sum) || throw(ArgumentError("weights cannot contain Inf or NaN values"))
     wv.values[i] = v
     wv.sum = sum
+    wv.positive = missing
     v
 end
 
+function Base.all(f::Base.Fix2{typeof(>=)}, wv::AbstractWeights)
+    if iszero(f.x)
+        if ismissing(wv.positive)
+            # sum is significantly faster than all when no entries are negative
+            wv.positive = sum(<(0), wv.values) == 0
+        end
+        return wv.positive
+    else
+        return all(f, wv.values)
+    end
+end
+
+function Base.any(f::Base.Fix2{typeof(<)}, wv::AbstractWeights)
+    if iszero(f.x)
+        if ismissing(wv.positive)
+            # sum is significantly faster than all when no entries are negative
+            wv.positive = sum(<(0), wv.values) == 0
+        end
+        return !wv.positive
+    else
+        return any(f, wv.values)
```

Comment:
add a test to cover this case.

Reply:
This is https://github.com/JuliaStats/StatsBase.jl/pull/834/files#r962207803. ;-)

Reply:
So why does coverage show that it is not covered?

Reply:
Because it was not applied before the last commit.